darts/tests/models/forecasting/test_regression_models.py

```diff
@@ -1909,6 +1935,308 @@ def test_component_specific_lags(self, config):
+
+            if "lags_future_covariates" not in list_lags:
+                # without future lags, all lags should be identical between shift and non-shifted model
+                np.testing.assert_almost_equal(
+                    pred_[-ocl_shifted:].all_values(copy=False),
+                    pred_shift_.all_values(copy=False),
+                )
+            else:
+                # without future lags, the shifted model also shifts future lags
+                with pytest.raises(AssertionError):
+                    np.testing.assert_almost_equal(
+                        pred_[-ocl_shifted:].all_values(copy=False),
+                        pred_shift_.all_values(copy=False),
+                    )
+
+            # with adjusted future lags, the models should be identical
+            np.testing.assert_almost_equal(
+                pred_[-ocl_shifted:].all_values(copy=False),
+                pred_shift_adj_.all_values(copy=False),
+            )
+
+    @pytest.mark.parametrize(
+        "config",
+        itertools.product(
+            [
+                {"lags": [-1, -3]},
+                {"lags_past_covariates": 2},
+                {"lags_future_covariates": [-2, -1]},
+                {"lags_future_covariates": [1, 2]},
+                {
+                    "lags": 5,
+                    "lags_past_covariates": [-3, -1],
+                },
+                {"lags": [-5, -4], "lags_future_covariates": [-2, 0, 1, 2]},
+                {
+                    "lags": 5,
+                    "lags_past_covariates": 4,
+                    "lags_future_covariates": [-3, 1],
+                },
+            ],
+            [3, 7, 10],
+        ),
+    )
+    def test_output_shift(self, config):
+        """Tests shifted output for shift smaller than, equal to, and larger than output_chunk_length."""
+        np.random.seed(0)
+        lags, shift = config
+        ocl = 7
+        series = tg.gaussian_timeseries(
+            length=28, start=pd.Timestamp("2000-01-01"), freq="d"
+        )
+
+        model_target_only = LinearRegressionModel(
+            lags=3,
+            output_chunk_length=ocl,
+            output_chunk_shift=shift,
+        )
+        model_target_only.fit(series)
+
+        # no auto-regression with shifted output
+        with pytest.raises(ValueError) as err:
+            _ = model_target_only.predict(n=ocl + 1)
+        assert str(err.value).startswith("Cannot perform auto-regression")
+
+        # pred starts with a shift
+        for ocl_test in [ocl - 1, ocl]:
+            pred = model_target_only.predict(n=ocl_test)
+            assert pred.start_time() == series.end_time() + (shift + 1) * series.freq
+            assert len(pred) == ocl_test
+            assert pred.freq == series.freq
+
+        # check that shifted output chunk results with encoders are the
+        # same as using identical covariates
+        model = LinearRegressionModel(
```

Comment:
This model is not used in the rest of the test (aside from generating the covariates dictionary), `model_enc_shift` and `model_fc_shift` are. Wouldn't it be simpler to just check the keys in the lags dictionary?